src/objectExplorer/objectExplorerService.ts

```diff
@@ -277,15 +308,19 @@ export class ObjectExplorerService {
                         let node = await promise;
                         // if password failed
                         if (!node) {
-                            const connectNode = new ConnectTreeNode(element);
-                            this._treeNodeToChildrenMap.set(element, [connectNode]);
-                            return [connectNode];
+                            // remove password if it's saved in the credential store
+                            let profile = element.connectionCredentials as IConnectionProfile;
+                            let password = await this._connectionManager.connectionStore.lookupPassword(profile);
+                            if (password) {
+                                await this.removeProfilePassword(profile);
+                                return this.handleNodeCreationFailure(element);
+                            } else {
+                                return this.handleNodePasswordFailure(element);
```

Comment:
These function names seem backwards to me - if we had a saved password we're calling handleNodeCreationFailure? But if we don't we call handleNodePasswordFailure? Is that intended?